app/validators/date_range_validator.rb

```diff
@@ -1,22 +1,30 @@
 class DateRangeValidator < ActiveModel::EachValidator
+  def initialize(**kwargs)
+    super(**kwargs)
+  end
+
   def validate_each(record, attribute, value)
-    unless range.cover?(value)
-      record.errors.add(attribute, :date_range, start_date: start_date_str, end_date: end_date_str)
-    end
+    return if value.nil? || range.cover?(value)
+
+    record.errors.add(attribute, :date_range, start_date: start_date_str, end_date: end_date_str)
   end
 
   private
 
   def start_date_str
+    return nil if date_format.nil?
```

Comment:
would we not want a default string to be returned ?  i.e. the date in a 'standard' format - having these values as nil might cause issues if you had a message in the translation file that needed them to be strings ?